### ISSUE TITLE
Bliss: change default splitting heuristic

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#1170]
`igraph_isomorphic` uses Bliss (except for special cases). Bliss has a parameter that affects performance: 

http://www.tcs.hut.fi/Software/bliss/doxy/classbliss_1_1Graph.html#a08da370e34106cd7db479eca7c7375cc

Currently, igraph uses "first".

I suggest changing this to "first largest" after having done some benchmarking for verification.

I did some benchmarking using Mathematica's `GraphData`, which is a large database of interesting _undirected_ graphs.

Taking all graphs with 80 or more vertices from this database (739 graphs in total as of Mathematica 12.0), here is the total running time in seconds for each heuristic:

First	32.1853
FirstLargest	5.6012
FirstMaximallyConnected	5.68259
FirstSmallestMaximallyConnected	5.69174
FirstLargestMaximallyConnected	5.56614

"FirstSmallest" is excluded because it didn't finish in more than half an hour.

The results above are a bit misleading. "First" is slow because it gets stuck on a _single_ graph, `{"DifferenceSetIncidence", {73, 64, 56}}`.   Let's exclude it and re-benchmark:

First	8.04303
FirstLargest	5.48274
FirstMaximallyConnected	5.89298
FirstSmallestMaximallyConnected	5.84067
FirstLargestMaximallyConnected	5.52655

As you can see, "First" is a bit slower, and the rest perform similarly, _on average_.

I propose switching the default heuristic to "FirstLargest" (`IGRAPH_BLISS_FL`).

----

### Caveats

I benchmarked the automorphism group computation. Canonical labelling should perform similarly, as it also computed the automorphism group.

Before doing this, I propose doing a similar benchmark on directed graphs. Does anyone have a good test collection?

Most graphs in this dataset are sparse. Histogram of densities and vertex counts:

![image](https://user-images.githubusercontent.com/1212871/57063251-e6042f80-6cc2-11e9-92ca-a91204a25b78.png)

![image](https://user-images.githubusercontent.com/1212871/57063265-f74d3c00-6cc2-11e9-82c9-8b15a517678d.png)

These are not random graphs. They are are all graphs with some special mathematical significance. In my opinion, that makes this a good benchmark set.

----

Timing vs vertex count:

![image](https://user-images.githubusercontent.com/1212871/57064920-274b0e00-6cc8-11e9-8539-28ca3b63c93d.png)

----

### Code

For reference, here's the benchmark code:

```
bgl = AssociationMap[GraphData, GraphData[80 ;;]];

bgl2 = AssociationMap[GraphData, 
   DeleteCases[
    GraphData[80 ;;], {"DifferenceSetIncidence", {73, 64, 56}}]
   ];

timings=<||>;
Function[sh,
  Echo[
    {
      sh, 
      First@Set[
         timings[sh],
         Timing[ First@Timing@IGBlissAutomorphismGroup[#,"SplittingHeuristics"->sh] & /@ bgl2 ]
      ]
    }
  ]
] /@ 
{"First",(*"FirstSmallest",*)"FirstLargest","FirstMaximallyConnected","FirstSmallestMaximallyConnected","FirstLargestMaximallyConnected"}
```

**Comment by szhorvat** [comment]
Here are results for a set of graphs (line graphs of [Hadamard graphs](http://mathworld.wolfram.com/HadamardGraph.html)) where "first" does particularly well compared to the other methods.

![image](https://user-images.githubusercontent.com/1212871/57072722-f4604480-6cde-11e9-9d00-daf976b7ea37.png)

**Comment by ntamas** [comment]
I trust your judgment here that it's worth switching the default heuristics on BLISS to "first largest"; "first" seems to perform better on a small subset of the graphs of interest only.

**Comment by szhorvat** [comment]
I am experimenting with automorphism groups at the moment. Let's give it some time, and see if problems pop up.

Let us also try to find good _directed_ datasets to test on.

If there are no problems, I'll submit a PR.

**Comment by szhorvat** [comment]
Bliss's default (the one used by the command line program) is `IGRAPH_BLISS_FSM`.

**Comment by szhorvat** [comment]
There is no choice of heuristic that is best for all cases. It looks like in IGraph/M I went with FirstLargest as the default. According to the above benchmarks, it does very well on many graphs. I'll change the C core of igraph to that as well.

There are two places this affects igraph:

 - In `igraph_isomorphic`
 - In the documentation of the Bliss functions, where we can make a suggestion. I'll also mention that Bliss's won default is FSM, not FL.

**Comment by szhorvat** [comment]
Fixed by 12c4634c550578e058a653b94462c5d89a1c4049
